src/app/core/components/header/user-panel/user-panel.component.ts

```diff
@@ -48,8 +48,10 @@ export class UserPanelComponent implements OnInit {
   private subscribeOnRouterEvents() {
     this.router.events
       .pipe(filter(e => e instanceof NavigationEnd))
-      .subscribe((navEnd: NavigationEnd) => {
-        this.showUserPanel = navEnd.url !== `/${AppRoutePath.Login}`;
-      })
+      .subscribe(({ url }: NavigationEnd) => this.checkIfShowUserPanel(url))
```

Comment:
you chould `unsubscribe` from this subscription anyway.
there are several ways to do so:
1. Hold a reference to subscription and in `ngOnDestroy` unsubscibe:
`this.someSub.unsubscribe`
2. pretty the same as in first options, but hold an array of subscriptions and then use some helper like: `unsubscribeAll(this.subs)`
3. use `take, first, last, taleUntil` operators
4. pretty the same but: use `this.destroy$ = new Subject()` -> `takeUntil(this.destroy$),` -> `ngOnDestroy() { this.destroy.next(); this.destroy.complete(); } `

Reply:
Ok, thanks. I haven't unsubscribed in cases where component is initiated only once. However, I realize that it is better to consider unsubscribing in any case.